### PR TITLE
feat: introduce dependency injection for storage and token management

### LINKED ISSUE #14: Discogs gen 2 sdk
Update the sdk to accept Memory storage like zustand

### PR DESCRIPTION
**Pull request by paprikaf**
Core:
- Implement dependency injection pattern for better testability
- Add interfaces for HTTP, OAuth, and Token management
- Create DefaultTokenManager implementation
- Improve error handling and types
- Add comprehensive test suite with mocks
- Move utilities to dedicated utils directory
- Restructure implementations and factories
- Remove FileSystemStorage from core package

Example:
- Add interactive OAuth flow examples
- Implement collection and search operation examples
- Add environment variable validation
- Structure examples by feature (auth/collection/search)
- Update configuration and dependencies
- Improve TypeScript configuration

Breaking Changes:
- Storage adapters now require async interface
- FileSystemStorage moved from core to example
- Token management interface updated

Closes #14: Discogs Gen 2 SDK update to support memory storage like Zustand.

### CODE REVIEW

**Comment by paprikaf** [comment]
@govi218  pr ready for you

File: core/src/auth/types.ts [hunk 1]
```diff
@@ -1,64 +1,32 @@
-export declare type RequestTokenResponse = {
-    oauthRequestToken: string;
-    oauthRequestTokenSecret: string;
+import { HttpClient } from 'src/interfaces/http';
+import { OAuthState, OAuthTokenPair } from '../interfaces/oauth';
+import { StorageAdapter } from 'src/interfaces/storage';
+
+export interface RequestTokenResponse {
```
Comment (govi218) [inline core/src/auth/types.ts:5]: these should be namespaced to Discogs (e.g. `DiscogsRequestTokenResponse`)

File: core/src/auth/types.ts [hunk 2]
```diff
@@ -1,64 +1,32 @@
...
+    port?: number;
+    host?: string;
+    path?: string;
+    timeout?: number;
+    customSuccessHtml?: string;
+}
+
+
+
+export { OAuthTokenPair } from '../interfaces/oauth';
```
Comment (govi218) [inline core/src/auth/types.ts:32]: ??

**Review by govi218 — COMMENTED** [review]
My main gripe with OOP is that this is a ton of glue code that we have to maintain now. Apart from that there are some minor changes.

**Comment by govi218** [comment]
In the future, we should discuss how we can contain PRs to reduce reviewer cognitive load; refer to this: https://betterprogramming.pub/how-to-create-pull-requests-which-are-a-pleasure-to-review-d4ef46c63d07

File: core/README.md [hunk 3]
```diff
@@ -1,56 +1,160 @@
-# Discogs SDK
+# @crate.ai/discogs-sdk
 
-The Discogs SDK is a library that uses the Discogs API to authenticate and access their data. Currently, the library supports the authentication flow, collection, search and retrieving the user's identity.
+A TypeScript SDK for the Discogs API with dependency injection support.
```
[the remark this answers is not in the document]
Comment (paprikaf) [reply core/README.md:3]: wow

**Comment by govi218** [comment]
open mouth maximum

File: core/README.md [hunk 4]
```diff
@@ -1,56 +1,160 @@
...
 
-# Getting Started
-1. sign in to discogs and go to [developer settings](https://www.discogs.com/settings/developers)
-2. click on "New App"
-3. fill out the form and click "Create App"
-4. Obtain your consumer key and secret from Discogs [here](https://www.discogs.com/settings/developers).
-5. Install the library using npm: `npm install @crate.ai/discogs-sdk`.
+## Features
```
[the remark this answers is not in the document]
Comment (paprikaf) [reply core/README.md:5]: add it youself

File: core/README.md [hunk 5]
```diff
@@ -1,56 +1,160 @@
...
+    query: 'Album Name',
+    type: 'release',
+    year: '1977',
+    format: 'album'
+});
+```
+
+## Examples
+
+For complete examples, check out our [example project](https://github.com/Crate-AI/discogs-sdk).
```
[the remark this answers is not in the document]
Comment (paprikaf) [reply core/README.md]: done

File: example/README.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Discogs SDK Examples
+
+This project demonstrates how to use the @crate.ai/discogs-sdk package with practical examples.
+
+## Setup
+
+1. Clone the repository
+2. Install dependencies:
```
[the remark this answers is not in the document]
Comment (paprikaf) [reply example/README.md:8]: for the example package that's all the readme you need buddy

**Pull request merged**